File: two_sum.go
func twoSum(nums []int, target int) []int {
	if len(nums) < 2 {
    	return nil
    }
	m := make(map[int]int)
	for index, value := range nums {
		m[value] = index
	}
	for index, value := range nums {
		key := target - value
		mm, _ := m[key]
		if mm > 0 {
			return []int{index, mm}
		}
	}
	return nil
}

File: TwoSum.swift
// https://leetcode.com/problems/two-sum/

class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        guard nums.count >= 2 else {
            return []
        }
        var map: [Int: Int] = [:]
        for (offset, value) in nums.enumerated() {
            let key = target - value
            if let i = map[key] {
                return [i, offset]
            }
            map[value] = offset
        }
        return []
    }
}